src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/personalrelationships/profiledetails/ContactPersonProfileDetailsResource.kt

```diff
@@ -0,0 +1,69 @@
+package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.personalrelationships.profiledetails
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.tags.Tag
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.config.BadRequestException
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.config.ErrorResponse
+
+@RestController
+@RequestMapping("/sync/contactperson/profile-details", produces = [MediaType.APPLICATION_JSON_VALUE])
+@Tag(name = "contact-person-migration-resource")
+@PreAuthorize("hasAnyRole('ROLE_MIGRATE_CONTACTPERSON', 'ROLE_MIGRATE_NOMIS_SYSCON')")
+class ContactPersonProfileDetailsResource(
+  private val syncService: ContactPersonProfileDetailsSyncService,
+) {
+  @PutMapping("/{prisonerNumber}/{bookingId}/{profileType}")
+  @Operation(
+    summary = "Synchronises a profile detail to DPS",
+    description = "Manually synchronises a profile detail to NOMIS. This is intended for use by developers to recover from errors. Requires role <b>MIGRATE_CONTACTPERSON</b> or <b>MIGRATE_NOMIS_SYSCON/b>",
```

Comment:
The main change is an endpoint for manually syncing to DPS for data fixes when things go wrong.